introduction-to-doctests-in-haskell/article.md

```diff
@@ -0,0 +1,289 @@
+# Introduction to Doctests in Haskell
+
+If you ever had to skim through the Haskell source code documentation, you might have noticed those lines with fancy `>>>` symbols at the beginning.
+They are called 'doctests' or 'doctest examples'.
+
+To put it simply, doctests are pieces of text within [Haddock comments](https://www.haskell.org/haddock/) that look like interactive Haskell sessions.
+The idea to search for and execute those sessions all over Haskell modules actually comes from the awesome Python's
+[`doctest`](https://en.wikipedia.org/wiki/Doctest) module.
+
+Documenting software is extremely challenging.
+Doctests make the documentation process not only pleasant and effective – it actually allows Haskell functions to speak for themselves – but
+also helps to get the most out of your testing efforts.
+
+## How to define a doctest?
+
+
+Let's look at an example of a doctest. 
+
+```haskell
```

Comment:
Want to perhaps take something from Haskell source? 

Like:

-- | @const x@ is a unary function which evaluates to @x@ for all inputs.
--
-- >>> const 42 "hello"
-- 42
--
-- >>> map (const 42) [0..3]
-- [42,42,42,42]

Here, the lack of comment and the <BLANKLINE> seems a bit confusing to me, personally.

Reply:
I've added this particular piece a bit higher, let me know if you don't like that.

Reply:
I do agree with the changes :)